### PR TITLE
Varying temperature cop

### PR DESCRIPTION
**Pull request by jimrojerTNO**
This PR adds functionality to have a varying COP for heat pumps when using varying network temperature with the carnot cop formulation from: https://industrialheatpumps.nl/english/operating_principle/cop_heat_pump/

### CODE REVIEW

File: src/rtctools_heat_network/esdl/esdl_heat_model.py [hunk 1]
```diff
@@ -645,6 +645,7 @@ def convert_heat_pump(self, asset: Asset) -> Tuple[Type[HeatPump], MODIFIERS]:
 
         modifiers = dict(
             COP=cop,
+            efficiency=asset.attributes["efficiency"] if asset.attributes["efficiency"] else 1.0,
```
Comment (FJanssen-TNO) [inline src/rtctools_heat_network/esdl/esdl_heat_model.py:648]: default value of the efficiency should actually be lower than 1.0 to represent reality. Typically the efficiency would be 50-80%. Setting it to 0.8 still results in a optimisitic but somewhat more realistic efficiency.

File: src/rtctools_heat_network/heat_physics_mixin.py [hunk 2]
```diff
@@ -2384,6 +2384,108 @@ class and the selected temperature in the network.
...
+                                    1.0 - self.state(f"{prim_ret_carrier}_{prim_ret_temp}")
+                                    if len(prim_ret_temps) > 0
+                                    else 0
+                                )
+                                efficiency = parameters[f"{hp}.efficiency"]
+                                t_cond = 273.15 + sec_sup_temp
+                                t_evap = 273.15 + prim_sup_temp
+
+                                cop_carnot = efficiency * t_cond / (t_cond - t_evap)
+                                selected = (
```
Comment (FJanssen-TNO) [inline src/rtctools_heat_network/heat_physics_mixin.py:2465]: add comment that selected is 0 when all the tempeartures are selected, just based on the names and not looking back how you configered them one would actually expect the opposite

**Review by FJanssen-TNO — COMMENTED** [review]
The set up seems appropriate and the added constraints are fine, only small comments on default values and description

File: src/rtctools_heat_network/esdl/esdl_heat_model.py (hunk 1 above)
Comment (jimrojerTNO) [reply src/rtctools_heat_network/esdl/esdl_heat_model.py:648]: I agree, I will actually set it to 50%

File: src/rtctools_heat_network/heat_physics_mixin.py (hunk 2 above)
Comment (jimrojerTNO) [reply src/rtctools_heat_network/heat_physics_mixin.py]: changed the name to not_selected

**Review by FJanssen-TNO — APPROVED** [review]

**Pull request merged**